algorithms/kernel/objective_function/mse/mse_dense_default_batch_impl.i

```diff
@@ -46,7 +46,9 @@ inline services::Status MSEKernel<algorithmFPType, method, cpu>::compute(Numeric
 {
     const size_t nDataRows = dataNT->getNumberOfRows();
     int result             = 0;
-    if (componentOfGradient || componentOfHessianDiagonal || componentOfProximalProjection)
+    if (componentOfGradient ||
+        (componentOfHessianDiagonal && (parameter->resultsToCompute & objective_function::componentOfHessianDiagonal)) ||
```

Comment:
`parameter->resultsToCompute & objective_function::componentOfHessianDiagonal`  can be computed once above and re-used as bool variable.

Reply:
it was fixed